django/db/models/query.py

```diff
@@ -321,6 +321,7 @@ def iterator(self):
         An iterator over the results from applying this QuerySet to the
         database.
         """
+        self.query.chunked_fetch = True
```

Comment:
We might want to avoid storing this flag in `self.query` as iterating over this queryset (invoking `self.__iter__()`) would also use `chunked_fetch`. As you've mentioned this doesn't change anything now since `self._fetch_all()` calls `iterator()` under the hood but I don't think we want to use server side cursors for every query.

I suggest me pass `chunked_fetch` as a parameter to `self._iterable_class()` and make `_fetch_all` uses `list(self._iterable_class(self))` instead of calling `iterator()`.

Passing the _chunked_ parameter on iterable instantiation could also us to define a `__del__` method on them to delete the named cursor in cases where the iterator is not completely exhausted but falls out of scope in autocommit mode.